contracts/SiphonDepositor.sol

```diff
@@ -0,0 +1,185 @@
+    IAuraLocker public immutable auraLocker;
+    uint256 public immutable pid;
+
+    /**
+     * @dev Penalty basis points 2500 == 25%
+     */
+    uint256 public immutable penaltyBp;
+
+    constructor(
+        IERC20 _lpToken,
+        IERC20 _crv,
+        IBooster _booster,
+        ICvx _cvx,
+        IrCvx _rCvx,
+        IAuraLocker _auraLocker,
+        uint256 _pid,
+        uint256 _penaltyBp
+    ) {
+        lpToken = _lpToken;
+        crv = _crv;
+        booster = _booster;
+        cvx = _cvx;
+        rCvx = _rCvx;
+        auraLocker = _auraLocker;
+        pid = _pid;
+        penaltyBp = _penaltyBp;
+    }
+
+    /**
+     * @dev deposit "lpTokens" into the booster
+     */
+    function deposit() external {
+        lpToken.approve(address(booster), type(uint256).max);
+        uint256 bal = lpToken.balanceOf(address(this));
+        booster.deposit(pid, bal, true);
+    }
+
+    /**
+     * @dev withdraw "lpTokens" from the booster
+     */
+    function withdraw(uint256 _amount) external {
+        booster.withdraw(pid, _amount);
+    }
+
+    /**
+     * @dev Siphon AURA tokens by depositing BAL into the Booster
+     *      and then calling earmark rewards which will send the BAL
+     *      to the siphon pools BaseRewardPool
+     * @param _amount Amount of BAL that is being bridged from the L2 to cover the
+     *                Incentive amount that was paid out.
+     *                We assume the total incentives that have been paid out are equal
+     *                to the MaxFees on the Booster which is 2500/10000 (25%)
+     */
+    function siphon(uint256 _amount) external {
+        // TODO: only callable by admin or lzEndpoint
+
+        uint256 amount = (_amount * 10000) / 2500;
+        uint256 bal = crv.balanceOf(address(this));
+        require(bal >= amount, "!balance");
+
+        // Transfer CRV to the booster and earmarkRewards
+        crv.transfer(address(booster), amount);
+        booster.earmarkRewards(pid);
+
+        // Mint rCvx at a rate of 1:1 rAURA:BALRewards
+        // TODO: send rAURA to the lzEndpoint (L2)
+        rCvx.mint(address(this), amount);
+    }
+
+    /**
+     * @dev Call getReward on the BaseRewardPool which will return the
+     *      BAL we previously depoisted minus the incentives that were paid
+     *      Along with a pro rata amount of AURA tokens
+     */
+    function getReward() external {
+        (, , , address crvRewards, , ) = booster.poolInfo(pid);
+        IBaseRewardPool(crvRewards).getReward(address(this), false);
+    }
+
+    /**
+     * @dev How much AURA you will get for rAURA
+     *      rAURA can be redeemed at a rate determined by the totalSupply
+     *      of rAURA and totalAmount of AURA the depositor has farmed.
+     */
+    function getAmountOut(uint256 _amount) public view returns (uint256) {
+        uint256 totalSupply = rCvx.totalSupply();
+        uint256 farmedTotal = cvx.balanceOf(address(this));
```

Comment:
WE could maybe put a hard cap on this of 3.8 cvx per rcvx. This would allow the pre-farming to give a bit of a buffer while the system matures